src/role_squadsiege.js

```diff
@@ -40,20 +40,22 @@ roles.squadsiege.preMove = function(creep, directions) {
 
   }
 
-  if (!creep.memory.initialized) {
-    Memory.squads[creep.memory.squad].siege[creep.id] = {};
-    creep.memory.initialized = true;
-  }
-  var squad = Memory.squads[creep.memory.squad];
-  if (squad.action == 'move') {
-    if (creep.room.name == squad.moveTarget) {
-      let nextExits = creep.room.find(creep.memory.routing.route[creep.memory.routing.routePos].exit);
-      let nextExit = nextExits[Math.floor(nextExits.length / 2)];
-      let range = creep.pos.getRangeTo(nextExit.x, nextExit.y);
-      if (range < 2) {
-        Memory.squads[creep.memory.squad].siege[creep.id].waiting = true;
-        creep.moveRandom();
-        return true;
+  if (creep.memory.squad) {
+    if (!creep.memory.initialized) {
+      Memory.squads[creep.memory.squad].siege[creep.id] = {};
+      creep.memory.initialized = true;
+    }
+    var squad = Memory.squads[creep.memory.squad];
+    if (squad.action == 'move') {
+      if (creep.room.name == squad.moveTarget) {
+        let nextExits = creep.room.find(creep.memory.routing.route[creep.memory.routing.routePos].exit);
+        let nextExit = nextExits[Math.floor(nextExits.length / 2)];
+        let range = creep.pos.getRangeTo(nextExit.x, nextExit.y);
+        if (range < 2) {
+          Memory.squads[creep.memory.squad].siege[creep.id].waiting = true;
+          creep.moveRandom();
```

Comment:
Hm, not sure for this case. If maybe even a `creep.moveRandom(true)` to stay on the path.

In the first version the squadsiege was flipping two to three positions before exiting the room until the squadsiege arrives, which looked promising.

Will have to redo this anyway.